Laboratory/C++/gen/jazz.hpp

```diff
@@ -17,25 +18,28 @@ struct Musician {
   std::string name;
   Instrument plays;
 
-  static std::unique_ptr<std::vector<uint8_t>> encode(const Musician& message) {
-    bebop::BebopWriter writer{};
+  static void encodeInto(const Musician& message, std::vector<uint8_t> &targetBuffer) {
+    ::bebop::Writer writer{targetBuffer};
     Musician::encodeInto(message, writer);
-    return writer.buffer();
   }
 
-  static void encodeInto(const Musician& message, bebop::BebopWriter& writer) {
+  static void encodeInto(const Musician& message, ::bebop::Writer& writer) {
     writer.writeString(message.name);
     writer.writeUint32(static_cast<uint32_t>(message.plays));
   }
 
-  static Musician decode(const uint8_t *buffer) {
+  static Musician decode(const uint8_t *sourceBuffer) {
     Musician result;
-    bebop::BebopReader reader{buffer};
-    Musician::decodeInto(result, reader);
+    Musician::decodeInto(sourceBuffer, result);
     return result;
   }
 
-  static void decodeInto(Musician& target, bebop::BebopReader& reader) {
+  static void decodeInto(const uint8_t *sourceBuffer, Musician& target) {
```

Comment:
Minor nit pick is that you have two different styles for `*` and `&` here (right align and left align). I would expect either:
```C++
static void decodeInto(const uint8_t* sourceBuffer, Musician& target)
```
or
```C++
static void decodeInto(const uint8_t *sourceBuffer, Musician &target)
```